readthedocs/builds/tasks.py

```diff
@@ -667,3 +667,74 @@ def send_webhook(self, webhook):
                 webhook_id=webhook.id,
                 webhook_url=webhook.url,
             )
+
+
+@app.task(queue="web")
+def check_and_disable_project_for_consecutive_failed_builds(project_slug, version_slug):
+    """
+    Check if a project has too many consecutive failed builds and disable it.
+
+    When a project has more than RTD_BUILDS_MAX_CONSECUTIVE_FAILURES consecutive failed builds
+    on the default version, we attach a notification to the project and disable builds (skip=True).
+    This helps reduce resource consumption from projects that are not being monitored.
+    """
+    from itertools import groupby
+
+    from readthedocs.builds.constants import BUILD_STATE_FINISHED
+    from readthedocs.projects.notifications import (
+        MESSAGE_PROJECT_BUILDS_DISABLED_DUE_TO_CONSECUTIVE_FAILURES,
+    )
+
+    # Skip disable builds for Read the Docs for Business
+    if settings.ALLOW_PRIVATE_REPOS:
+        return
```

Comment:
We can probably move this check to the task invocation, so we don't create an extra task for nothing.